src/os_esp32.cc

```diff
@@ -200,7 +200,7 @@ class ConditionVariable {
 
     mutex_->unlock();
 
-#ifdef CONFIG_IDF_TARGET_ESP32C3
+#ifndef CONFIG_IDF_TARGET_ESP32
```

Comment:
I don't understand this. Do we not have uint32 defined?

Reply:
I think using following judgement maybe more direct:
```
#if defined(CONFIG_IDF_TARGET_ESP32C3) || (CONFIG_IDF_TARGET_ESP32S2) || (CONFIG_IDF_TARGET_ESP32S3)
```
This can help make it easy to add more SoC support to toit. By a way, why not use `uint32_t` directly, this is defined in <stdint.h>.

Reply:
Changed to `uint32_t` and removed the `#ifdef`.